app/src/main/java/com/x8bit/bitwarden/data/autofill/accessibility/manager/AccessibilityFieldManagerImpl.kt

```diff
@@ -0,0 +1,82 @@
+package com.x8bit.bitwarden.data.autofill.accessibility.manager
+
+import android.net.Uri
+import android.util.Log
+import android.view.accessibility.AccessibilityNodeInfo
+import com.x8bit.bitwarden.data.autofill.accessibility.util.getKnownUsernameFieldNull
+import com.x8bit.bitwarden.data.autofill.accessibility.util.isUsername
+
+private const val MAX_NODE_COUNT: Int = 100
+private const val MAX_RECURSION_DEPTH: Int = 100
+
+/**
+ * The default implementation for the [AccessibilityFieldManager].
+ */
+class AccessibilityFieldManagerImpl : AccessibilityFieldManager {
+    override fun findAccessibilityNodeInfos(
+        rootNode: AccessibilityNodeInfo,
+        recursionDepth: Int,
+        predicate: (AccessibilityNodeInfo) -> Boolean,
+    ): List<AccessibilityNodeInfo> {
+        if (predicate(rootNode)) return listOf(rootNode)
+        if (recursionDepth >= MAX_RECURSION_DEPTH) return emptyList()
+        val childNodeCount = rootNode.childCount - 1
+        if (childNodeCount > MAX_NODE_COUNT) log(message = "Too many child iterations.")
+        return (0..childNodeCount.coerceAtMost(maximumValue = MAX_NODE_COUNT)).flatMap {
+            val childNode = rootNode.getChild(it) ?: return@flatMap emptyList()
+            if (childNode.hashCode() == this.hashCode()) {
+                log(message = "Child node is the same as parent for some reason.")
```

Comment:
There are two logging spots in this function, I am not actually sure if we want them.

Reply:
I've made these debug only, but we do want to keep them.